exchanges/bitfinex/bitfinex_test.go

```diff
@@ -951,3 +953,37 @@ func TestGetDepositAddress(t *testing.T) {
 		}
 	}
 }
+
+// TestWsAuth dials websocket, sends login request.
+func TestWsAuth(t *testing.T) {
+	b.SetDefaults()
+	TestSetup(t)
+	if !b.Websocket.IsEnabled() && !b.AuthenticatedAPISupport || !areTestAPIKeysSet() {
+		t.Skip(exchange.WebsocketNotEnabled)
+	}
+	var err error
+	var dialer websocket.Dialer
+	b.WebsocketConn, _, err = dialer.Dial(b.Websocket.GetWebsocketURL(),
+		http.Header{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	b.Websocket.DataHandler = make(chan interface{}, 999)
```

Comment:
Yeah 999 is definitely excessive! I'll reduce those by a couple of digits 👍

Regarding a shared const are though, isnt it that you cannot have a shared package with tests like websocket_shared_test.go since test files aren't compiled? I can put some consts in say exchange_websocket.go and reference those in tests since there's only two items to be shared, otherwise can just have some consts at the top of the test files which would still clean things better than the way I've done it inline.

Reply:
Ah that's right, since we can't export things from _test.go files we would need a test helper package for exchanges which will only be compiled in for tests by the exchanges using those helpers so there's no risk of bloating release code with test default vars.